### ISSUE TITLE
Error on roblox

### ISSUE THREAD

**Issue by pokecosimo** [SpaceGT/Anti-AFK#15]
Hello, i use this anti afk for roblox and gives me this error 
![281533962-0c130d58-c2ce-42ab-8070-5d6987bb066a](https://github.com/SpaceGT/Anti-AFK/assets/96430819/4d071af0-e2c0-4b89-8b25-df59b2f54100)

**Comment by SpaceGT** [comment]
Interesting.
This is the second time someone has encountered this issue.
I will try and debug it; does it happen often?

**Comment by pokecosimo** [comment]
> Interesting. This is the second time someone has encountered this issue. I will try and debug it; does it happen often?

yeah. everytime

**Comment by pokecosimo** [comment]
> Interesting. This is the second time someone has encountered this issue. I will try and debug it; does it happen often?

did u fix it?

**Comment by towermom9** [comment]
yo I got the same bug after I updated windows. I think it's somehow related?

**Comment by towermom9** [comment]
@pokecosimo have you tried running it with admin privileges? (right click then run as administrator)

**Comment by pokecosimo** [comment]
> @pokecosimo have you tried running it with admin privileges? (right click then run as administrator)

yes

**Comment by SpaceGT** [comment]
Hello there o/
I wasn't able to reproduce this on my computer.
I'll try updating windows, but that tends to take a while.

**Comment by SpaceGT** [comment]
Well that didn't work :(

It's clear the system the script is running on definitely has a role in whether this issue arises.
Perhaps slower computers are not able to activate the window in time?
We can get around this by using `WinWaitActive()` after a window is activated.

Can one of you please test if this fixes it?

Here's the modified function:
```
resetTimer(targetWindow, resetAction, DenyInput)
{
    if (!WinExist("A") || (WinGetClass("A") = "WorkerW" || WinGetClass("A") = "Progman"))
    {
        WinActivate("ahk_id " targetWindow)
        WinWaitActive("ahk_id " targetWindow)
    }

    if (WinActive("ahk_id " targetWindow))
    {
        resetAction()
        return
    }

    if (DenyInput && A_IsAdmin)
        BlockInput("On")

    oldWindow := WinGetID("A")

    WinSetTransparent(0, "ahk_id " targetWindow)
    WinActivate("ahk_id " targetWindow)
    WinWaitActive("ahk_id " targetWindow)

    resetAction()

    WinMoveBottom("ahk_id " targetWindow)
    WinSetTransparent("OFF", "ahk_id " targetWindow)

    WinActivate("ahk_id " oldWindow)
    WinWaitActive("ahk_id " oldWindow)

    if (DenyInput && A_IsAdmin)
        BlockInput("Off")
}
```

**Comment by towermom9** [comment]
yeah I think it's related to the system. because I had the problem with my Intel gen 4 laptop. but using it on my gen 12 laptop it runs fine.

**Comment by pokecosimo** [comment]
> Well that didn't work :(
> 
> It's clear the system the script is running on definitely has a role in whether this issue arises. Perhaps slower computers are not able to activate the window in time? We can get around this by using `WinWaitActive()` after a window is activated.
> 
> Can one of you please test if this fixes it?
> 
> Here's the modified function:
> 
> ```
> resetTimer(targetWindow, resetAction, DenyInput)
> {
>     if (!WinExist("A") || (WinGetClass("A") = "WorkerW" || WinGetClass("A") = "Progman"))
>     {
>         WinActivate("ahk_id " targetWindow)
>         WinWaitActive("ahk_id " targetWindow)
>     }
> 
>     if (WinActive("ahk_id " targetWindow))
>     {
>         resetAction()
>         return
>     }
> 
>     if (DenyInput && A_IsAdmin)
>         BlockInput("On")
> 
>     oldWindow := WinGetID("A")
> 
>     WinSetTransparent(0, "ahk_id " targetWindow)
>     WinActivate("ahk_id " targetWindow)
>     WinWaitActive("ahk_id " targetWindow)
> 
>     resetAction()
> 
>     WinMoveBottom("ahk_id " targetWindow)
>     WinSetTransparent("OFF", "ahk_id " targetWindow)
> 
>     WinActivate("ahk_id " oldWindow)
>     WinWaitActive("ahk_id " oldWindow)
> 
>     if (DenyInput && A_IsAdmin)
>         BlockInput("Off")
> }
> ```

i will test

**Comment by pokecosimo** [comment]
> Well that didn't work :(
> 
> It's clear the system the script is running on definitely has a role in whether this issue arises. Perhaps slower computers are not able to activate the window in time? We can get around this by using `WinWaitActive()` after a window is activated.
> 
> Can one of you please test if this fixes it?
> 
> Here's the modified function:
> 
> ```
> resetTimer(targetWindow, resetAction, DenyInput)
> {
>     if (!WinExist("A") || (WinGetClass("A") = "WorkerW" || WinGetClass("A") = "Progman"))
>     {
>         WinActivate("ahk_id " targetWindow)
>         WinWaitActive("ahk_id " targetWindow)
>     }
> 
>     if (WinActive("ahk_id " targetWindow))
>     {
>         resetAction()
>         return
>     }
> 
>     if (DenyInput && A_IsAdmin)
>         BlockInput("On")
> 
>     oldWindow := WinGetID("A")
> 
>     WinSetTransparent(0, "ahk_id " targetWindow)
>     WinActivate("ahk_id " targetWindow)
>     WinWaitActive("ahk_id " targetWindow)
> 
>     resetAction()
> 
>     WinMoveBottom("ahk_id " targetWindow)
>     WinSetTransparent("OFF", "ahk_id " targetWindow)
> 
>     WinActivate("ahk_id " oldWindow)
>     WinWaitActive("ahk_id " oldWindow)
> 
>     if (DenyInput && A_IsAdmin)
>         BlockInput("Off")
> }
> ```

im testing it

**Comment by pokecosimo** [comment]
> Well that didn't work :(
> 
> It's clear the system the script is running on definitely has a role in whether this issue arises. Perhaps slower computers are not able to activate the window in time? We can get around this by using `WinWaitActive()` after a window is activated.
> 
> Can one of you please test if this fixes it?
> 
> Here's the modified function:
> 
> ```
> resetTimer(targetWindow, resetAction, DenyInput)
> {
>     if (!WinExist("A") || (WinGetClass("A") = "WorkerW" || WinGetClass("A") = "Progman"))
>     {
>         WinActivate("ahk_id " targetWindow)
>         WinWaitActive("ahk_id " targetWindow)
>     }
> 
>     if (WinActive("ahk_id " targetWindow))
>     {
>         resetAction()
>         return
>     }
> 
>     if (DenyInput && A_IsAdmin)
>         BlockInput("On")
> 
>     oldWindow := WinGetID("A")
> 
>     WinSetTransparent(0, "ahk_id " targetWindow)
>     WinActivate("ahk_id " targetWindow)
>     WinWaitActive("ahk_id " targetWindow)
> 
>     resetAction()
> 
>     WinMoveBottom("ahk_id " targetWindow)
>     WinSetTransparent("OFF", "ahk_id " targetWindow)
> 
>     WinActivate("ahk_id " oldWindow)
>     WinWaitActive("ahk_id " oldWindow)
> 
>     if (DenyInput && A_IsAdmin)
>         BlockInput("Off")
> }
> ```

it seams working

**Comment by SpaceGT** [comment]
Glad to hear it!
I'll push out the fix then :)

**Comment by SpaceGT** [comment]
This should now be fixed in f8afd99e007788f21e4d6a927005a00fb01d1141
